api/envoy/api/v2/route/route.proto

```diff
@@ -437,6 +439,22 @@ message RouteAction {
     google.protobuf.Duration per_try_timeout = 3 [(gogoproto.stdduration) = true];
   }
 
+  // Specifies the idle timeout for the route. If not specified, there is no
+  // stream level timeout. This is distinct to :ref:`timeout
+  // <envoy_api_field_route.RouteAction.timeout>`, which provides an upper bound
+  // on the upstream response time; :ref:`idle_timeout
+  // <envoy_api_field_route.RouteAction.idle_timeout>` instead bounds the amount
+  // of time the request's stream may be idle.
+  //
+  // After header decoding, and assuming that this is not a WebSocket upgrade,
+  // the idle timeout will apply on downstream and upstream request events. Each
+  // time an encode/decode event for headers or data is processed for the
+  // stream, the timer will be reset. If the timeout fires, the stream is
+  // terminated with a 408 Request Timeout error code if no upstream response
+  // header has been received, otherwise a stream reset occurs.
+  google.protobuf.Duration idle_timeout = 24
```

Comment:
Are we concerned that people are going to want different idle timeouts for decode than for encode? I could see this happening w/ asymmetric streaming APIs. I agree that resetting the idle timeout in both cases means that a single value can be used, I'm just concerned that we might end up adding different timeouts later. Do we think this can happen or can we say it will never be needed? If we think it can happen should we future proof the API somehow to make this not awkward? Would rather not have to go through a deprecation cycle to deal with that.

Reply:
Yes, that sounds correct.

Reply:
OK sgtm.